Make the emoji clickable in all pipleine jobs slack notification

aaradhak opened the pull request:
Changes in the pipeline job slack notifications where the emoji itself is clickable and the build number is printed only once.

dustymabe commented on jobs/check-aws-regions.Jenkinsfile:
```diff
@@ -26,7 +26,7 @@ cosaPod(serviceAccount: "jenkins"){
     if (disabled_regions != "None") {
         warn("Disabled AWS regions detected: ${disabled_regions}")
-        pipeutils.trySlackSend(message: ":aws: aws-regions-disabled :jenkins:<${env.BUILD_URL}|${env.BUILD_NUMBER}> ${disabled_regions}")
+        pipeutils.trySlackSend(message: ":aws: aws-regions-disabled :jenkins:<${env.BUILD_URL}|${env.BUILD_NUMBER}> :ocean:<${env.RUN_DISPLAY_URL}|${env.BUILD_NUMBER}> ${disabled_regions}")
```
Thanks for this. I've been thinking it would be nice to make the emoji itself clickable so we only have to include the build # once (we would need to apply this to all jobs and not just this new one).

Also I think it would be better to have the first bit of output from this job be the name of the job `check-aws-regions`. 

```suggestion
        pipeutils.trySlackSend(message: ":aws: check-aws-regions #${env.BUILD_NUMBER} <${env.BUILD_URL}|:jenkins:> <${env.RUN_DISPLAY_URL}|:ocean:> detected disabled regions: ${disabled_regions}")
```

aaradhak replied:
sure,  will make this change to all the jobs

marmijo commented on jobs/check-aws-regions.Jenkinsfile:
```diff
@@ -26,7 +26,7 @@ cosaPod(serviceAccount: "jenkins"){
     if (disabled_regions != "None") {
         warn("Disabled AWS regions detected: ${disabled_regions}")
-        pipeutils.trySlackSend(message: ":aws: aws-regions-disabled :jenkins:<${env.BUILD_URL}|${env.BUILD_NUMBER}> ${disabled_regions}")
+        pipeutils.trySlackSend(message: ":aws: check-aws-regions #${env.BUILD_NUMBER} <${env.BUILD_URL}|:jenkins:> <${env.RUN_DISPLAY_URL}|:ocean:> detected disabled regions: ${disabled_regions}")
```
Was this tested? When we added the blue ocean link to the other notifications, I couldn't find any documentation that supported  this. I hope it works!

dustymabe replied:
I suggested it in https://github.com/coreos/fedora-coreos-pipeline/pull/899#discussion_r1276742315 but I have not tested it.

Which part are you worried about working? the hyperlinking of an emoji?

marmijo replied:
Yes, I didn't think slack allowed this. It's worth a try!

aaradhak replied:
I am not able to test it on the stage pipeline. Probably we can merge this and see how it works and then make the changes to other jobs?

marmijo replied:
Yeah that works for me. I wasn't too happy with repeating the build number in the other jobs so this would be ideal!

marmijo commented on jobs/bump-lockfile.Jenkinsfile:
```diff
@@ -297,7 +297,7 @@ lock(resource: "bump-${params.STREAM}") {
     } finally {
         if (currentBuild.result != 'SUCCESS') {
-            pipeutils.trySlackSend(message: ":jenkins:<${env.BUILD_URL}|bump-lockfile ${env.BUILD_NUMBER} (${params.STREAM})> :ocean:<${env.RUN_DISPLAY_URL}|${env.BUILD_NUMBER}>")
+            pipeutils.trySlackSend(message: "bump-lockfile #${env.BUILD_NUMBER} <${env.BUILD_URL}|:jenkins: (${params.STREAM})> <${env.RUN_DISPLAY_URL}|:ocean:>")
```
```suggestion
            pipeutils.trySlackSend(message: "bump-lockfile #${env.BUILD_NUMBER} <${env.BUILD_URL}|:jenkins:> (${params.STREAM})> <${env.RUN_DISPLAY_URL}|:ocean:>")
```

Missing the `>` to close out the jenkins hyperlink

aaradhak commented:
I would need to make one more change in the code.

marmijo reviewed (commented):
Looks great! This should also probably be updated in the `build`, `build-arch`, `build-cosa`, and `build-fcos-buildroot` jobs as well.

dustymabe commented on jobs/check-aws-regions.Jenkinsfile:
```diff
@@ -25,8 +25,9 @@ cosaPod(serviceAccount: "jenkins"){
         disabled_regions = shwrapCapture("ore aws list-regions --disabled")
     }
     if (disabled_regions != "None") {
+        echo "echo regions: ${disabled_regions}"
```
pretty sure we should delete this line

aaradhak replied:
ah yea, i ll get rid of this. I was trying to test something.

dustymabe commented on jobs/check-aws-regions.Jenkinsfile:
```diff
@@ -25,8 +25,9 @@ cosaPod(serviceAccount: "jenkins"){
         disabled_regions = shwrapCapture("ore aws list-regions --disabled")
     }
     if (disabled_regions != "None") {
```
I think this should be:

```
if (disabled_regions != "") {
```

dustymabe replied:
and maybe let's put that in a separate commit since we are fixing a bug.

dustymabe reviewed (changes requested):
LGTM once we fix https://github.com/coreos/fedora-coreos-pipeline/pull/899/commits/677ee0c90f6c87615f3692ed1fdc452e9f04ab5e#r1278032102

dustymabe commented:
The 2nd commit needs a more complete commit title and body. Otherwise everything LGTM.

aaradhak commented:
> The 2nd commit needs a more complete commit title and body. Otherwise everything LGTM.

Yes,  I pushed it before I added the description. I have added the description for it.

aaradhak commented:
> The 2nd commit needs a more complete commit title and body. Otherwise everything LGTM.

Hope you are referring to this commit - https://github.com/coreos/fedora-coreos-pipeline/pull/899/commits/8b00a7930869603949621a45cd72b5f1a42946f7

I have made the changes now. I had pushed it before adding the description.

dustymabe commented:
```This branch has conflicts that must be resolved```

aaradhak commented:
ok 

> `This branch has conflicts that must be resolved`
> 
> You need to rebase on top of latest `main`.

ok , I ll rebase it
